packages/core/src/plugins/__tests__/SegmentDestination.test.ts

```diff
@@ -419,4 +418,86 @@ describe('SegmentDestination', () => {
       }
     );
   });
+  describe('getEndpoint', () => {
+    it('should throw an error when proxy ends with "/" and useSegmentEndpoints is true', () => {
+      const plugin = new SegmentDestination();
+      // Set up config dynamically
+      const config = {
+        ...clientArgs.config,
+        useSegmentEndpoints: true,
+        proxy: 'example.com/v1/',
+      };
+      plugin.analytics = new SegmentClient({
+        ...clientArgs,
+        config,
+      });
+
+      // Create a spy for the private method
+      const spy = jest.spyOn(
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        Object.getPrototypeOf(plugin) as any,
+        'getEndpoint'
+      );
+      // Expect the private method to throw an error
+      expect(() => plugin['getEndpoint']()).toThrow(
+        'Invalid proxy url has been passed'
+      );
+      // Ensure the spy was called
+      expect(spy).toHaveBeenCalled();
+    });
+  });
+  it('should throw an error when proxy contains a query parameter', () => {
```

Comment:
not ture. if `example.com/v1?params=xx` is passed and useSegmentEndpoints = false, this is resolved to `example.com/v1?params=xx` which is validate.